### PR TITLE
Samples for search

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/search/azure-search-documents/azure/search/documents/_service/aio/_indexers_client.py [hunk 1]
```diff
@@ -39,21 +39,21 @@ def __init__(self, endpoint, credential, **kwargs):
             endpoint=endpoint, sdk_moniker=SDK_MONIKER, **kwargs
         )  # type: _SearchServiceClient
 
-    def __enter__(self):
+    async def __aenter__(self):
```
Comment (rakshith91) [inline sdk/search/azure-search-documents/azure/search/documents/_service/aio/_indexers_client.py:42]: messed up aio in indexer :(

**Comment by bryevdv** [comment]
@rakshith91  can you update the samples readme as well:

https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/search/azure-search-documents/samples

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    fields = [
+        SimpleField(name="hotelId", type=edm.String, key=True),
+        SimpleField(name="baseRate", type=edm.Double)
+    ]
+    index = Index(name=index_name, fields=fields)
+    ind_client = service_client.get_indexes_client()
+    async with ind_client:
+        index = await ind_client.create_index(index)
+
+    # [START create_indexer_async]
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:57]: I don't see a sample snippet for creating a data source, which is required to even use an indexer. We should either include the lines above about creating a data source as part of this one, or at least have snippets called out in the data source-specific sample file below (which I didn't see).

Comment (bryevdv) [reply sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:57]: @heaths  isn't that up on line 38?

Comment (heaths) [reply sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:57]: Yes, I meant including in a snippet block using your START/END constructs.

Comment (rakshith91) [reply sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:57]: We already have snippets for datasource in a different file - I don't understand the rationale behind adding snippets for creating datasource in "indexers_client" documentation. If we do it for datasource, should we also include creating an index in the START/END construct?

Comment (heaths) [reply sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:57]: An indexer requires a data source. It's sole purpose is to index documents from a data source into an index, optionally applying AI enrichments.

**Comment by bryevdv** [comment]
@heaths @rakshith91 can we get this merged today to support the UX study, and make any necessary issues for other things to follow-up on?

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+service_client = SearchServiceClient(service_endpoint, AzureKeyCredential(key))
+indexers_client = service_client.get_indexers_client()
+
+async def create_indexer():
+    # create an index
+    index_name = "hotels"
+    fields = [
+        SimpleField(name="hotelId", type=edm.String, key=True),
+        SimpleField(name="baseRate", type=edm.Double)
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:42]: Just for what looks like a more complete (enough) example, maybe add a `hotelName` that is a `SearchableField` with a few bells and whistles. As a consumer, this index would give me nothing useful (I'd see a base rate, but for whom?).

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    ]
+    index = Index(name=index_name, fields=fields)
+    ind_client = service_client.get_indexes_client()
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:45]: Eventually, the method name should comply with the guidelines, i.e. that the model name is part of not only the client name ({Model}Client), but the method is get_{Model}Client (granted, in snake case idiomatic of python). Might consider doing before the UX study so, if people say "that's too long", we can take that back to archboard.

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    async with ind_client:
+        index = await ind_client.create_index(index)
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:47]: The methods - now on sub-clients - should not repeat the model. I.e. just `create`. Same as below for other sub-clients.

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    # create a datasource
+    ds_client = service_client.get_datasources_client()
+    credentials = DataSourceCredentials(connection_string=connection_string)
+    container = DataContainer(name='searchcontainer')
+    ds = DataSource(name="async-indexer-datasource", type="azureblob", credentials=credentials, container=container)
+    async with ds_client:
+        data_source = await ds_client.create_datasource(ds)
+
+    # create an indexer
+    indexer = Indexer(name="async-sample-indexer", data_source_name="async-indexer-datasource", target_index_name="hotels")
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:59]: `SearchIndexer`. See #10860.

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    ]
+    index = Index(name=index_name, fields=fields)
+    ind_client = service_client.get_indexes_client()
+    async with ind_client:
+        index = await ind_client.create_index(index)
+
+    # [START create_indexer_async]
+    # create a datasource
+    ds_client = service_client.get_datasources_client()
+    credentials = DataSourceCredentials(connection_string=connection_string)
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:52]: This connection string should just go onto the `SearchIndexerDataSource`. See #10860.

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    container = DataContainer(name='searchcontainer')
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:53]: `SearchIndexerDataContainer`. See #10860.

File: sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    ds = DataSource(name="async-indexer-datasource", type="azureblob", credentials=credentials, container=container)
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py:54]: `SearchIndexerDataSource`. See #10860.

File: sdk/search/azure-search-documents/samples/files/patient_data.json [hunk 10]
```diff
@@ -0,0 +1,187 @@
+{
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/files/patient_data.json:1]: Did you already have samples using medical data? Seems most others have been using the hotels sample. As long as you're consistent, though. If this is the first of new samples, you might consider a consistent model (i.e. hotels) for similarity across languages' samples.

File: sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+        SimpleField(name="contractedfromwhichpatientsuspected", type=edm.String),
+        SimpleField(name="currentstatus", type=edm.String, filterable=True),
+        SimpleField(name="dateannounced", type=edm.String),
+        SimpleField(name="detectedcity", type=edm.String),
+        SimpleField(name="detecteddistrict", type=edm.String),
+        SimpleField(name="detectedstate", type=edm.String),
+        SimpleField(name="estimatedonsetdate", type=edm.String),
+        SimpleField(name="gender", type=edm.String, filterable=True),
+        SimpleField(name="nationality", type=edm.String),
+        SimpleField(name="notes", type=edm.String),
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py:64]: Why all SimpleFields? At leaset for this one, using a SearchableField not only provides some diversity but would also set a better recommendation for such fields that people might want to use analyzers.

File: sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+        SimpleField(name="patientnumber", type=edm.String, key=True, filterable=True),
+        SimpleField(name="statecode", type=edm.String),
+        SimpleField(name="typeoftransmission", type=edm.String),
+    ]
+    cors_options = CorsOptions(allowed_origins=["*"], max_age_in_seconds=60)
+
+    # pass in the name, fields and cors options and create the index
+    index = Index(
```
Comment (heaths) [inline sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py:72]: `SearchIndex`. See #10860.

File: sdk/search/azure-search-documents/samples/files/patient_data.json (hunk 10 above)
Comment (heaths) [reply sdk/search/azure-search-documents/samples/files/patient_data.json:1]: Having skimmed the contents as well, I'm a little worried the topic of conversation might be triggering to people who have been impacted by this tragedy. Hotels seem pretty safe.

File: sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py (hunk 12 above)
Comment (bryevdv) [reply sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py:72]: Yes I think it is best to merge this PR, then do the updates in the other PR?

**Comment by bryevdv** [comment]
Regarding #10860 I had imagined merging this PR first, and then continuing any updates in #10860 so we can't make those changes in this PR.

**Comment by rakshith91** [comment]
will work on renaming in a different pr

File: sdk/search/azure-search-documents/samples/files/patient_data.json (hunk 10 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/samples/files/patient_data.json:1]: changed it to use hotel data - agree with patient data being sensitive - yes i already had the data for a different project i've been working on

File: sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py (hunk 11 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py:64]: added

File: sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py (hunk 12 above)
Comment (rakshith91) [reply sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py:67]: 10860 should be done in a different PR

**Pull request merged**